cli/flox/doc/manifest.toml.md

```diff
@@ -0,0 +1,310 @@
+pip = { pkg-path = "python310Packages.pip" }
+```
+Flox will use the first format by default when automatically editing
+the manifest.
+
+```{.include}
+./include/package-names.md
+```
+
+### Package descriptors
+
+Each entry in the `[install]` table is a key-value pair.
+The key in the key-value pair (e.g. `ripgrep`, `pip`) is referred to as an
+"install ID",
+and represents the name by which you will refer to a particular package e.g.
+if you wanted to uninstall or upgrade the package.
+Install IDs are inferred from the last attribute in the pkg-path,
+but may also be specified either at install-time via the `-i` option
+or interactively via `flox edit`.
+
+The value in the key-value pair is called a "package descriptor".
+A package is specified by a number of available options which are separate
+from the install ID,
+so you are free to change them independently of one another.
+This allows you to change package details while keeping a stable install ID,
+for example upgrading from `gcc.pkg-path = "gcc12"` to
+`gcc.pkg-path = "gcc13"`.
+
+The descriptor options allow you to specify in detail the package to install.
+The full list of descriptor options are shown below:
+```
+Descriptor ::= {
+  name               = null | <STRING>
+, optional           = null | <BOOL>
+, pkg-group          = null | <STRING>
+, version            = null | <STRING>
+, semver             = null | <STRING>
+, systems            = null | [<STRING>, ...]
+, pkg-path           = null | <STRING> | [<STRING>, ...]
+, abs-path           = null | <STRING> | [<STRING>, ...]
+, priority           = null | <INT>
+}
+```
+
+None of these options are required,
+and leaving them unset instructs Flox to simply find the best match for the
+package name and latest version given the install ID.
+In order to leave all options unset,
+simply use an empty package descriptor:
+
+```
+ripgrep = {}
+```
+
+By specifying some of these options you create a set of requirements that the
+installed program must satisfy,
+otherwise installation will fail.
+The most common option will likely be the `semver` option,
+which allows you to specify a semantic version range.
+
+Each option is described below:
+
+`name`
+:   Matches either the last attribute of the `pkg-path` or the package metadata
+    fields `name` or `pname` as set by the catalog.
+    This option is mutually exclusive with the `pkg-path` and `abs-path`
+    options.
+    You shouldn't need to use this option and should instead prefer the
+    `pkg-path` option.
+
+`optional`
+:   Marks this package as only an optional requirement for the environment.
+    By default an environment will fail to build if a specified package can't
+    be found in the catalog.
+    However, some packages are platform specific and will never be found in the
+    catalog for some systems.
+    Thus, the way you mark a package as platform specific is by setting
+    `opitonal = true` and using the `systems` option to list the systems on
```

Comment:
blocking: you can either set `optional` or add to the `systems` list and `options.systems`, but you don't have to do both

Reply:
Fixed